File: jupyter2/ps/build_system/readxml2/ml/readxml2/bin/readxml.ml
let parse filename =
  (* read file *)
  let ic = open_in filename in
  (* read the whole contents *)
  let contents = really_input_string ic (in_channel_length ic) in
  (* close the file *)
  let _ = close_in ic in
  (* print the contents *)
  Xml.parse_string contents

let rec height x = match x with
  | Xml.Element (_, _, children)
    -> 1 + List.fold_left (fun acc child -> max acc (height child)) 0 children
  | _ -> 0

File: jupyter2/ps/build_system/readxml2/rs/readxml2/src/readxml.rs
use std::fs;
extern crate minidom;
use minidom::Element;

pub fn parse(filename: &str) -> Element {
    let dat = fs::read_to_string(filename).unwrap();
    dat.parse().unwrap()
}

pub fn height(e: &Element) -> usize {
    if e.children().count() == 0 {
	1
    } else {
	1 + e.children().map(|c| height(c)).max().unwrap()
    }
}
